src/Cli/dotnet/commands/dotnet-workload/clean/WorkloadCleanCommand.cs

```diff
@@ -0,0 +1,137 @@
+        private readonly ReleaseVersion _sdkVersion;
+        private readonly IInstaller _workloadInstaller;
+        private readonly IWorkloadResolver _workloadResolver;
+
+        public WorkloadCleanCommand(
+            ParseResult parseResult,
+            IReporter? reporter = null,
+            IWorkloadResolver? workloadResolver = null,
+            string? dotnetDir = null,
+            string? version = null,
+            string? userProfileDir = null
+            ) : base(parseResult, reporter: reporter)
+        {
+            _cleanAll = parseResult.GetValue(WorkloadCleanCommandParser.CleanAllOption);
+
+            _dotnetPath = dotnetDir ?? Path.GetDirectoryName(Environment.ProcessPath);
+            if (_dotnetPath == null)
+            {
+                throw new GracefulException(String.Format(LocalizableStrings.InvalidWorkloadProcessPath, Environment.ProcessPath ?? "null"));
+            }
+
+            _userProfileDir = userProfileDir ?? CliFolderPathCalculator.DotnetUserProfileFolderPath;
+
+            _sdkVersion = WorkloadOptionsExtensions.GetValidatedSdkVersion(parseResult.GetValue(WorkloadUninstallCommandParser.VersionOption), version, _dotnetPath, userProfileDir, true);
+            var sdkFeatureBand = new SdkFeatureBand(_sdkVersion);
+
+            var workloadManifestProvider = new SdkDirectoryWorkloadManifestProvider(_dotnetPath, _sdkVersion.ToString(), _userProfileDir);
+            _workloadResolver = workloadResolver ?? WorkloadResolver.Create(workloadManifestProvider, _dotnetPath, _sdkVersion.ToString(), _userProfileDir);
+            _workloadInstaller = WorkloadInstallerFactory.GetWorkloadInstaller(Reporter, sdkFeatureBand, _workloadResolver, Verbosity, _userProfileDir, VerifySignatures, PackageDownloader, _dotnetPath);
+            
+        }
+
+        public override int Execute()
+        {
+            ExecuteGarbageCollection();
+            return 0;
+        }
+
+        private void ExecuteGarbageCollection()
+        {
+            if (_cleanAll)
+            {
+                _workloadInstaller.GarbageCollectInstalledWorkloadPacks(cleanAllPacks: true);
+            }
+            else
+            {
+                _workloadInstaller.GarbageCollectInstalledWorkloadPacks();
+            }
+
+            DisplayUninstallableVSWorkloads();
+        }
+
+        /// <summary>
+        /// Print VS Workloads with the same machine arch which can't be uninstalled through the SDK CLI to increase user awareness that they must uninstall via VS.
+        /// </summary>
+        private void DisplayUninstallableVSWorkloads()
+        {
+#if !DOT_NET_BUILD_FROM_SOURCE
+            if (OperatingSystem.IsWindows())
+            {
+                // All VS Workloads should have a corresponding MSI based SDK. This means we can pull all of the VS SDK feature bands using MSI/VS related registry keys.
+                var installedSDKVersionsWithPotentialVSRecords = MsiInstallerBase.GetInstalledSdkVersions();
```

Comment:
That's true. But I think it's better to show it in both cases. I don't think the average user will be aware of the difference of whether their SDK is in file-based or MSI based mode. And they might be confused as to why things from VS weren't removed if their machine is still in a bad state after cleaning with `--all`. IMO this messaging from us reduces the confusion, but it may be annoying to see the message pop up for every VS workload if you don't intend to mess with them.

Is the main concern about accessing the registry (elevation concerns) in a file-based mode? I (think*) registry reads are generally [permitted](https://superuser.com/questions/308693/if-i-am-not-an-administrator-but-i-want-to-read-the-registry-what-is-easiest-wa#:~:text=Non%2Dadministrators%20have%20the%20same,Open%20the%20Windows%20Start%20Menu) regardless of user-privelleges. Please let me know if you disagree.

Reply:
We decided to limit this to MSI because another concern is that MSI doesnt clean up file based and vice versa and we don't want to overlap those two worlds. Also, custom dotnet directory worlds should ideally not overlap either, you don't want to affect, say, another sdk pinned by a global.json